paddle/fluid/distributed/collective/ProcessGroupGloo.cc

```diff
@@ -293,6 +293,14 @@ std::shared_ptr<ProcessGroup::Task> ProcessGroupGloo::AllReduce(
     std::vector<phi::DenseTensor>& inputs,
     std::vector<phi::DenseTensor>& outputs,
     const AllreduceOptions& opts) {
+  return AllReduce(inputs, outputs, opts, true);
```

Comment:
Gloo supports async function now?

Reply:
No, the gloo backend should be rewritten to support async. Since we already change the base interface passing sync_op to c++.